source/module_basis/module_nao/hydrogen_radials.cpp

```diff
@@ -0,0 +1,412 @@
+                std::pow(n_, 4)*
+                static_cast<double>(this->assoc_laguerre_.factorial(n_ + l_)) /
+                std::pow(a0, 3)
+            );
+            for(int ir = 0; ir != ngrid; ++ir)
+            {
+                // Bohr radius is 1.0
+                double rho = 2.0 * rgrid[ir] * charge / n_ / a0;
+                rvalue[ir] = norm_factor * pow(rho, l_) * exp(-rho/2.0) * this->assoc_laguerre_.value(
+                    n_, l_, rho
+                );
+            }
+            // what izeta, symbol and itype should be?
+            chi_[ichi].build(l_, true, ngrid, rgrid, rvalue, 0, n_ - 1 - l_, "", itype_, false);
+            chi_[ichi].normalize();
+            ++ichi;
+        }
+    }
+
+    delete[] rvalue;
+    delete[] rgrid;
+}
+*/
+std::vector<double> HydrogenRadials::generate_hydrogen_radial_segment(const double charge,
+                                                                      const int n,
+                                                                      const int l,
+                                                                      const double rmin,
+                                                                      const double rmax,
+                                                                      const double dr,
+                                                                      const int rank,
+                                                                      std::ofstream* ptr_log)
+{
+    double a0 = 1.0; // Bohr radius
+    int ngrid = static_cast<int>((rmax - rmin) / dr) + 1;
+    std::vector<double> rgrid(ngrid);
+    std::vector<double> rvalue(ngrid);
+
+    // initialize value for rgrid
+    for(int ir = 0; ir != ngrid; ++ir)
+    {
+        rgrid[ir] = rmin + ir * dr;
+    }
+
+    double norm_factor = sqrt(
+        4.0*std::pow(charge, 3)*
+        static_cast<double>(this->assoc_laguerre_.factorial(n - l - 1)) /
+        std::pow(double(n), 4) / 
+        static_cast<double>(this->assoc_laguerre_.factorial(n + l)) /
+        std::pow(a0, 3)
+    );
+
+    for(int ir = 0; ir != ngrid; ++ir)
+    {
+        // Bohr radius is 1.0
+        double rho = 2.0 * rgrid[ir] * charge / n / a0;
+        rvalue[ir] = norm_factor * std::pow(rho, l) * exp(-rho/2.0) * this->assoc_laguerre_.value(
+            n, l, rho
+        );
+    }
+    return rvalue;
+}
+
+double HydrogenRadials::radial_norm(const std::vector<double> rgrid,
+                                    const std::vector<double> rvalue)
+{
+    std::vector<double> integrand(rvalue.size());
+    for(int ir = 0; ir != rvalue.size(); ++ir)
+    {
+        integrand[ir] = rvalue[ir] * rvalue[ir] * rgrid[ir] * rgrid[ir];
+    }
+    double dr = rgrid[1] - rgrid[0];
+    double norm = ModuleBase::Integral::simpson(rvalue.size(), integrand.data(), dr);
+    norm = sqrt(norm);
+    return norm;
+}
+
+double HydrogenRadials::generate_hydrogen_radial_toconv(const double charge,
```

Comment:
wrap functions at least above two, generate hydrogen-like orbitals to convergence